flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/functions/SystemFunctionUtils.java

```diff
@@ -184,6 +231,53 @@ public static int timestampDiff(String symbol, long fromDate, long toDate) {
         }
     }
 
+    public static TimestampData timestampadd(
+            String timeintervalunit, int interval, LocalZonedTimestampData timepoint) {
+        return timestampadd(timeintervalunit, interval, timepoint.getEpochMillisecond());
+    }
+
+    public static TimestampData timestampadd(
+            String timeintervalunit, int interval, ZonedTimestampData timepoint) {
+        return timestampadd(timeintervalunit, interval, timepoint.getMillisecond());
+    }
+
+    public static TimestampData timestampadd(
+            String timeintervalunit, int interval, TimestampData timepoint) {
+        return timestampadd(timeintervalunit, interval, timepoint.getMillisecond());
+    }
+
+    public static TimestampData timestampadd(
+            String timeintervalunit, int interval, long timepoint) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(new Date(timepoint));
+        int field;
+        switch (timeintervalunit) {
+            case "SECOND":
+                field = Calendar.SECOND;
+                break;
+            case "MINUTE":
+                field = Calendar.MINUTE;
+                break;
+            case "HOUR":
+                field = Calendar.HOUR;
+                break;
+            case "DAY":
+                field = Calendar.DAY_OF_YEAR;
+                break;
+            case "MONTH":
+                field = Calendar.MONTH;
+                break;
+            case "YEAR":
+                field = Calendar.YEAR;
+                break;
+            default:
+                LOG.error("Unsupported timestamp add: {}", timeintervalunit);
```

Comment:
I think it's not very meaningful to log here since it was evaluated in Janino engine.